packages/universal-sql/src/calculateScore.js

```diff
@@ -0,0 +1,118 @@
+/* // generate with DESCRIBE SELECT column_type FROM test_all_types();
+type DuckDBColumnType =
+	| `STRUCT(${string})`
+	| `${string}[]`
+	| 'BIGINT'
+	| 'BIT'
+	| 'BOOLEAN'
+	| 'BLOB'
+	| 'DATE'
+	| 'DOUBLE'
+	| `DECIMAL(${number},${number})`
+	| 'HUGEINT'
+	| 'INTEGER'
+	| 'INTERVAL'
+	| 'FLOAT'
+	| 'SMALLINT'
+	| 'TIME'
+	| 'TIMESTAMP'
+	| 'TIMESTAMP_S'
+	| 'TIMESTAMP_MS'
+	| 'TIMESTAMP_NS'
+	| 'TIME WITH TIME ZONE'
+	| 'TIMESTAMP WITH TIME ZONE'
+	| 'TINYINT'
+	| 'UBIGINT'
+	| 'UINTEGER'
+	| 'USMALLINT'
+	| 'UTINYINT'
+	| 'UUID'
+	| 'VARCHAR';
+*/
+
+/**
+ * @param {string} column_type
+ * @returns {column_type is `STRUCT(${string})` | `${string}[]`}
+ */
+function isObjectType(column_type) {
+	return column_type.startsWith('STRUCT') || column_type.endsWith('[]');
+}
+
+/**
+ * @param {string} column_type
+ * @returns {column_type is `DECIMAL(${number},${number})`}
+ */
+function isDecimalType(column_type) {
+	return column_type.startsWith('DECIMAL');
+}
+
+/**
+ *
+ * @param {string} column_type
+ * @returns {number}
+ */
+function columnTypeToScore(column_type) {
+	const NUMBER = 12; // 8 bytes in a double + 4 bytes heap overhead?
+	const BOOLEAN = 4; // booleans are stored as 4 byte integers
+	const STRING = 2 * 15; // assume 15 character string
```

Comment:
Does the varchar size get exposed? That might be a little more reliable 

This works if not

Reply:
You mean something like the max sized string in the column? I don't think so